### PR TITLE
chore: PR title check ensures uppercase is used in first character

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Making use of a config in the existing tool we are using: https://github.com/amannn/action-semantic-pull-request/tree/main?tab=readme-ov-file#configuration

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

**Comment by AgustinBettati** [comment]
verified `Validate PR title` check fails when defining lower case in subject of title.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -63,24 +63,24 @@ For more explained information about plugin override check [Development Override
...
-- `ci: description of the PR`: The commit makes changes to continuous integration or continuous delivery scripts or configuration files (this correlates with Minor in Semantic Versioning).
-- `revert: description of the PR`: The commit reverts one or more commits that were previously included in the product, but were accidentally merged or serious issues were discovered that required their removal from the main branch (this correlates with Minor in Semantic Versioning).
-- `style: description of the PR`: The commit updates or reformats the style of the source code, but does not otherwise change the product implementation (this correlates with Minor in Semantic Versioning).
-- `feat: description of the PR`: a commit of the type feat introduces a new feature to the codebase (this correlates with MINOR in Semantic Versioning).
-- `deprecate: description of the PR`: The commit deprecates existing functionality, but does not remove it from the product (this correlates with MINOR in Semantic Versioning).
+We use [*Conventional Commits*](https://www.conventionalcommits.org/), enforcing the subject of the title to start with a uppercase character:
```
Comment (marcosuma) [inline CONTRIBUTING.md]: Would it make sense to mention that we also use third-person approach? Maybe a bit unrelated with the PR title but it's literally one sentence more here.

**Review by marcosuma — APPROVED** [review]

**Review by lantoli — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (andreaangiolillo) [inline CONTRIBUTING.md]: I am not sure why we are enforcing this 🤔 What is the value in the uppercase?

Comment (AgustinBettati) [reply CONTRIBUTING.md]: we have inconsistencies in our PR titles, one of them being starting with upper or lower case in the subject of the titles. This leads to manual adjustments and reviews when generating the changelog (example: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1768#discussion_r1431247961).

Comment (AgustinBettati) [reply CONTRIBUTING.md]: 👍 Included an additional sentence.

Comment (andreaangiolillo) [reply CONTRIBUTING.md]: thanks for the clarification!

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**